clvm/core_ops.py

```diff
@@ -20,41 +21,41 @@ def op_if(args):
     return IF_COST, r.first()
 
 
-def op_cons(args):
+def op_cons(args: SExp):
     if args.list_len() != 2:
         raise EvalError("c takes exactly 2 arguments", args)
     return CONS_COST, args.first().cons(args.rest().first())
 
 
-def op_first(args):
+def op_first(args: SExp):
     if args.list_len() != 1:
         raise EvalError("f takes exactly 1 argument", args)
     return FIRST_COST, args.first().first()
 
 
-def op_rest(args):
+def op_rest(args: SExp):
     if args.list_len() != 1:
         raise EvalError("r takes exactly 1 argument", args)
     return REST_COST, args.first().rest()
 
 
-def op_listp(args):
+def op_listp(args: SExp):
     if args.list_len() != 1:
         raise EvalError("l takes exactly 1 argument", args)
     return LISTP_COST, args.true if args.first().listp() else args.false
 
 
-def op_raise(args):
+def op_raise(args: SExp):
     raise EvalError("clvm raise", args)
 
 
-def op_eq(args):
+def op_eq(args: SExp):
     if args.list_len() != 2:
         raise EvalError("= takes exactly 2 arguments", args)
     a0 = args.first()
     a1 = args.rest().first()
-    if a0.pair or a1.pair:
-        raise EvalError("= on list", a0 if a0.pair else a1)
+    if a0.as_pair() or a1.as_pair():
```

Comment:
I think we should require that `SExp` also implement the `CLVMObject` protocol, ie. respond to `.pair` and `.atom` which makes this change unnecessary.

Reply:
I think it's problematic to have `SExp` also implement the `CLVMObject` protocol, as we can end up nesting them which makes things a bit more confused.

I'm fine with `SExp` having a different API, but I don't see a good reason to have both `as_pair()` *and* just `pair`